### PR TITLE
Feature/tdr 1/phpunit8 upgrade

### PR DESCRIPTION
**Pull request by Yury-Razhkou**
related to: https://oat-sa.atlassian.net/browse/TDR-1
depends on: https://github.com/oat-sa/generis/pull/751
Upgrade to phpunit8

### CODE REVIEW

File: test/integration/ServiceTest.php [hunk 1]
```diff
@@ -124,7 +124,7 @@ public function testAbstractService()
 
         //get the diff between the class and the subclass
         $diffs = $this->taoService->getPropertyDiff($testModelClass, $generisResourceClass);
-        $this->assertInternalType('array', $diffs);
+        $this->assertIsArray( $diffs);
```
Comment (bartlomiejmarszal) [inline test/integration/ServiceTest.php]:
```suggestion
        $this->assertIsArray($diffs);
```

File: test/integration/ServiceTest.php [hunk 2]
```diff
@@ -162,7 +162,7 @@ public function testAbstractService()
             $testClass = $itemClass;
         }
         $foundProp = $this->taoService->getClazzProperties($testClass);
-        $this->assertInternalType('array', $foundProp);
+        $this->assertIsArray( $foundProp);
```
Comment (bartlomiejmarszal) [inline test/integration/ServiceTest.php]:
```suggestion
        $this->assertIsArray($foundProp);
```

File: test/integration/controllerMap/FactoryTest.php [hunk 3]
```diff
@@ -165,10 +166,11 @@ public function testGetControllerDescription($controllerClassName)
      * @dataProvider missingControllerActionNameProvider
      * @param string $controllerClassName
      * @param string $actionName
-     * @expectedException oat\tao\model\controllerMap\ActionNotFoundException
+     * @throws ActionNotFoundException
```
Comment (bartlomiejmarszal) [inline test/integration/controllerMap/FactoryTest.php]:
```suggestion
```

File: test/integration/user/import/OntologyUserMapperTest.php [hunk 4]
```diff
@@ -34,7 +35,7 @@ class OntologyUserMapperTest extends GenerisPhpUnitTestRunner
      * @param $schema
      * @param $data
      * @param $expected
-     * @throws \Exception
+     * @throws Exception
```
Comment (bartlomiejmarszal) [inline test/integration/user/import/OntologyUserMapperTest.php]:
```suggestion
```

File: test/integration/user/import/OntologyUserMapperTest.php [hunk 5]
```diff
@@ -56,7 +57,7 @@ public function testMapUserWithSuccess($schema, $data, $expected)
      *
      * @dataProvider provideInsufficientDataExample
      *
-     * @throws \Exception
+     * @throws Exception
```
Comment (bartlomiejmarszal) [inline test/integration/user/import/OntologyUserMapperTest.php]:
```suggestion
```

File: test/integration/user/import/OntologyUserMapperTest.php [hunk 6]
```diff
@@ -73,12 +74,11 @@ public function testMapMandatoryShouldFail($schema, $data, $expected)
      * @param $expected
      *
      * @dataProvider provideEmptyFieldDataExample
-     * @expectedException \oat\tao\model\import\service\MandatoryFieldException
-     *
-     * @throws \Exception
+     * @throws Exception
```
Comment (bartlomiejmarszal) [inline test/integration/user/import/OntologyUserMapperTest.php]:
```suggestion
```

File: test/integration/user/import/RdsUserImportServiceTest.php [hunk 7]
```diff
@@ -31,7 +32,7 @@ class RdsUserImportServiceTest extends TestCase
     /**
      * @dataProvider provideMapperProperties
      * @param $data
-     * @throws \Exception
+     * @throws Exception
```
Comment (bartlomiejmarszal) [inline test/integration/user/import/RdsUserImportServiceTest.php]:
```suggestion
```

File: test/integration/user/import/RdsUserImportServiceTest.php [hunk 8]
```diff
@@ -47,11 +48,11 @@ public function testImport($data)
 
     /**
      * @dataProvider provideMapperProperties
-     * @expectedException \Exception
-     * @throws \Exception
+     * @throws Exception
```
Comment (bartlomiejmarszal) [inline test/integration/user/import/RdsUserImportServiceTest.php]:
```suggestion
```

File: test/unit/http/StreamRangeTest.php [hunk 9]
```diff
@@ -46,10 +47,13 @@ public function testConstruct($stream, $rangeValue, $first, $last)
 
     /**
      * @dataProvider wrongRangesProvider
-     * @expectedException \oat\tao\model\stream\StreamRangeException
+     * @param $stream
+     * @param $rangeValue
+     * @throws StreamRangeException
```
Comment (bartlomiejmarszal) [inline test/unit/http/StreamRangeTest.php]:
```suggestion
```

File: test/unit/lock/SimpleLockTest.php [hunk 10]
```diff
@@ -49,10 +50,10 @@ public function testConstruct()
         $owner = $this->prophesize('core_kernel_classes_Resource');
         $owner->getUri()->willReturn('#ownerUri');
         $lock = new SimpleLock($resource->reveal(), $owner->reveal(), 'epoch');
-        
+        $this->assertTrue(true);
```
Comment (bartlomiejmarszal) [inline test/unit/lock/SimpleLockTest.php]:
```suggestion
```

File: test/unit/model/taskQueue/TaskLog/CategorizedStatusTest.php [hunk 11]
```diff
@@ -21,13 +21,14 @@
...
 
+use Exception;
 use oat\tao\model\taskQueue\TaskLog\CategorizedStatus;
 use oat\generis\test\TestCase;
 
 class CategorizedStatusTest extends TestCase
 {
     /**
-     * @throws \Exception
+     * @throws Exception
```
Comment (bartlomiejmarszal) [inline test/unit/model/taskQueue/TaskLog/CategorizedStatusTest.php]:
```suggestion
     * @throws Exception
```

File: test/unit/model/taskQueue/TaskLog/CategorizedStatusTest.php [hunk 12]
```diff
@@ -36,16 +37,16 @@ public function testCreateWithValidStatus()
     }
 
     /**
-     * @expectedException \Exception
-     * @throws \Exception
+     * @throws Exception
```
Comment (bartlomiejmarszal) [inline test/unit/model/taskQueue/TaskLog/CategorizedStatusTest.php]:
```suggestion
```

File: test/unit/webhooks/WebhookEventsServiceTest.php [hunk 13]
```diff
@@ -223,8 +223,6 @@ public function testHandleEventNotSerializable()
         /** @noinspection PhpParamsInspection */
         $event = $this->createMock(Event::class);
         $event->method('getName')->willReturn('TestEvent');
-        $event->method('getWebhookEventName')->willReturn('WhTestEvent');
```
Comment (bartlomiejmarszal) [inline test/unit/webhooks/WebhookEventsServiceTest.php]: Why did you change that? Is this causing errors? Is this because of phpUnit upgrade?

File: test/unit/webhooks/task/JsonWebhookPayloadFactoryTest.php [hunk 14]
```diff
@@ -50,6 +50,7 @@ public function testCreatePayloadSchema()
             }
             $this->fail(implode(PHP_EOL, $validationErrors));
         }
+        $this->assertTrue(true);
```
Comment (bartlomiejmarszal) [inline test/unit/webhooks/task/JsonWebhookPayloadFactoryTest.php]:
```suggestion
```

Comment (Yury-Razhkou) [reply test/unit/webhooks/task/JsonWebhookPayloadFactoryTest.php]: AssertTrue added to remove risky label. This test in some cases do not perform assertions and risky label remove code coverage.

File: test/unit/webhooks/WebhookEventsServiceTest.php (hunk 13 above)
Comment (Yury-Razhkou) [reply test/unit/webhooks/WebhookEventsServiceTest.php]: Passed `Event` do not have such methods. And this method never called. But mocking it is incorrect.

Comment (bartlomiejmarszal) [reply test/unit/webhooks/WebhookEventsServiceTest.php]: Ok. It's not in the scope of this PR. Let's move this to different PR.

Comment (Yury-Razhkou) [reply test/unit/webhooks/WebhookEventsServiceTest.php]: Test won't work with this line

Comment (bartlomiejmarszal) [reply test/unit/webhooks/WebhookEventsServiceTest.php]: This is why when we extract this test to another PR we have to ensure it will be merged before this PR.

Comment (bartlomiejmarszal) [reply test/unit/webhooks/WebhookEventsServiceTest.php]: But in order to keep this PR consistent, we should keep only PHPUnit 8 upgrade changes. 
You can remove this change, create another PR for it and if you really want you can cherry-pick commit with this change.

Comment (Yury-Razhkou) [reply test/unit/webhooks/WebhookEventsServiceTest.php]: Removing this line related to upgrade. This line works fine on previous versions of phpunit and causes an error with phpunit8. It not just removing line because it is unnecessary here.
